tests/testsuite/build.rs

```diff
@@ -6407,3 +6408,44 @@ fn renamed_uplifted_artifact_remains_unmodified_after_rebuild() {
     let not_the_same = !same_file::is_same_file(bin, renamed_bin).unwrap();
     assert!(not_the_same, "renamed uplifted artifact must be unmodified");
 }
+
+#[cargo_test(nightly, reason = "debug options stabilized in 1.70")]
+fn debug_options_valid() {
+    use serde::Serialize;
+    use TomlDebugInfo::*;
+
+    for option in [
+        TomlDebugInfo::None,
+        LineDirectivesOnly,
+        LineTablesOnly,
+        Limited,
+        Full,
```

Comment:
Would it be possible to just use the strings here?  The important part here is to assert that certain strings get processed and passed-through appropriately to rustc. If at some point in the future the enum code is changed, that would also implicitly change this test, which means the test might not catch any incorrect changes.

For example, this could be something like:

```rust
for (option, cli) in [
    ("line-directives-only", "line-directives-only"),
    ("line-tables-only", "line-tables-only"),
    ("none", "0"),
    ("limited", "1"),
    ("full", "2"),
] {
```

and then in the test, it should check that the argument is getting passed to rustc correctly. Something roughly like:

```rust
p.cargo("build -v")
    .with_stderr_contains(&format!("[RUNNING] `rustc [..]-C debuginfo={cli}"))
    .run();
```

That way the test is double-checking that everything is getting passed through correctly, and any changes to cargo that break this flow would trip this test.

Reply:
Sure thing, done :)